src/CreatePhotosSlideshow/CreatePhotosSlideshow.tsx

```diff
@@ -96,41 +127,45 @@ const CreatePhotosSlideshow = (): JSX.Element => {
       setProgress(parseFloat((ratio * 100).toFixed(1)));
     });
 
-    for (const index of orderedItems) {
-      ffmpeg.FS(
-        'writeFile',
-        getCleanName(files[index]),
-        new Uint8Array(correctedFileArrayBuffer[index])
-      );
-    }
+    const inputPaths: Array<string> = [];
+    orderedItems.forEach(async (itemPosition, index) => {
+      const fileName = getCleanName(files[itemPosition].name);
+      inputPaths.push(`file ${fileName}\nduration 2`);
+      ffmpeg.FS('writeFile', fileName, await fetchFile(files[itemPosition]));
 
-    const ffconcatPath = 'slides.ffconcat';
-    const concatFileLines = ['ffconcat version 1.0']
-      .concat(
-        orderedItems.map((index) => {
-          const slideDuration = 1;
-          return `file ${getCleanName(
-            files[index]
-          )}\nduration ${slideDuration}`;
-        })
-      )
-      .join('\n');
+      // add the last file into the txt this is a quricky bug in ffmpeg
+      const isLastFile = index === orderedItems.length - 1;
+      if (isLastFile) {
+        inputPaths.push(`file ${fileName}`);
+      }
```

Comment:
Like it needs to be added twice?

Reply:
Concat Demuxer section here https://trac.ffmpeg.org/wiki/Slideshow mentioned it